dashboard/modules/job/common.py

```diff
@@ -127,6 +130,9 @@ def __post_init__(self):
     def to_json(self) -> Dict[str, Any]:
         """Convert this object to a JSON-serializable dictionary.
 
+        Note that the runtime_env field is converted to a JSON-serialized string
```

Comment:
nit: Possible to keep the naming consistent across the 2? Or we could use a map field for `JobsAPIInfo`?

Reply:
More discussion on the naming here: https://github.com/ray-project/ray/pull/31046#discussion_r1088053459 I don't feel strongly either way.
I think defining a `proto` for RuntimeEnv is out of scope of this PR, it's not a simple map<string, string> unfortunately.